src/jdbc/cubrid/jdbc/jci/UStmtCache.java

```diff
@@ -81,24 +81,32 @@ void clear() {
 		}
 	}
 
-	int remove_expired_res(long checkTime) {
+	int remove_expired_res(long checkTime, UUrlCache uc) {
 		UResCache rc;
+		UResCache victim = null;
+		int v_idx = -1;
 
 		for (int i = 0; i < res_cache_remove_list.size(); i++) {
 			rc = res_cache_remove_list.get(i);
 			if (rc.isExpired(checkTime)) {
-				res_cache_table.remove(rc.key);
+				if (victim == null || rc.getCacheTime() < victim.getCacheTime()) {
+					victim = rc;
+					v_idx = i;
+				}
+			}
+		}
 
-				synchronized (res_cache_remove_list) {
-				    	UResCache lastObj = res_cache_remove_list
-							.remove(res_cache_remove_list.size() - 1);
-					if (i < res_cache_remove_list.size()) {
-						res_cache_remove_list.set(i, lastObj);
-						i--;
-					}
+		if (victim != null) {
+			uc.addCacheSize(-victim.getCacheSize());
+			synchronized (res_cache_remove_list) {
+				res_cache_table.remove(victim.key);
+				UResCache lastObj = res_cache_remove_list.remove(res_cache_remove_list.size() - 1);
+				if (v_idx < res_cache_remove_list.size()) {
+					res_cache_remove_list.set(v_idx, lastObj);
 				}
 			}
 		}
+
 		return res_cache_remove_list.size();
```

Comment:
Why only the last selected victim is removed?

Reply:
for simplifying LRU, for each statement cache check victims and remove a victim because the result cache is distributed to several statement cache.

Reply:
Do you mean that the last index in res_cache_remove_list is the least recently used?

Reply:
by the victim selection algorithm, it picks the least serverTime of result-caches.

Reply:
Ok, I undetstood.